.github/workflows/release.yml

```diff
@@ -0,0 +1,127 @@
+name: Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      module:
+        description: "Maven (Sub)module to Release (e.g.,
+                  keycloak-extensions,
+                  authenticator-required-action,
+                  keycloak-extension-rest,
+                  user-management).
+                  Parent is the default"
+        required: false
+        default: "parent"
+      releaseVersion:
+        description: "Version to use for release."
+        required: true
+        default: "X.Y.Z"
+      developmentVersion:
+        description: "Version to use for next development cycle."
+        required: true
+        default: "X.Y.Z-SNAPSHOT"
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Configure Git User
+        run: |
+          git config user.email "actions@github.com"
+          git config user.name "GitHub Actions"
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v4
+        with:
+          distribution: 'temurin'
+          java-version: 17
+          cache: 'maven'
+
+      - name: Install Parent POM (If necessary locally)
+        run: mvn install -N || echo "Parent-POM already exists or is getting downloaded from repo."
+
+
+      - name: Determine Build Scope
+        id: determine_scope
+        run: |
+          if [[ "${{ github.event.inputs.module }}" == "parent" ]]; then
+            echo "BUILD_MODULE=parent" >> $GITHUB_ENV
+            echo "BUILD_ARGS=" >> $GITHUB_ENV
+          else
+            echo "BUILD_MODULE=${{ github.event.inputs.module }}" >> $GITHUB_ENV
+            echo "BUILD_ARGS=-pl ${{ github.event.inputs.module }} -am" >> $GITHUB_ENV
```

Comment:
The `--also-make | -am` will also build modules that this module depends on. These may be snapshot versions.
As we are in a release workflow we need to ensure that no snapshot versions are referenced.